net.solarnetwork.node.datum.energymeter.mock/src/net/solarnetwork/node/datum/energymeter/mock/MockEnergyMeterDatumSource.java

```diff
@@ -206,6 +279,9 @@ public void setMessageSource(MessageSource messageSource) {
 		results.add(new BasicTextFieldSettingSpecifier("frequency", defaults.frequency));
 		results.add(new BasicTextFieldSettingSpecifier("resistance", defaults.resistance));
 		results.add(new BasicTextFieldSettingSpecifier("inductance", defaults.inductance));
+		results.add(new BasicToggleSettingSpecifier("randomness", defaults.randomness));
+		results.add(new BasicTextFieldSettingSpecifier("voltdev", defaults.voltDeviation));
+		results.add(new BasicTextFieldSettingSpecifier("freqdev", defaults.freqDeviation));
```

Comment:
When creating `SettingSpecifier` instances, here is where you'd convert non-string types to strings. So assuming we had  a `float frequency` class field, I'd expect to see

```java
results.add(new BasicTextFieldSettingSpecifier("frequency", String.valueOf(defaults.frequency));
```

or possibly a formatted value like

```java
results.add(new BasicTextFieldSettingSpecifier("frequency", String.format("%.1f", defaults.frequency));
```